src/library/scala/util/matching/Regex.scala

```diff
@@ -116,29 +118,41 @@ import java.util.regex.{ Pattern, Matcher }
  *  while (mi.hasNext) {
  *    val d = mi.next
  *    if (mi.group(1).toInt < 1960) println(s"$d: An oldie but goodie.")
+ *  }
  *  }}}
  *
- *  Note that `findAllIn` finds matches that don't overlap. (See [[findAllIn]] for more examples.)
+ *  Although the `MatchIterator` returned by `findAllIn` is used like any `Iterator`,
+ *  with alternating calls to `hasNext` and `next`, `hasNext` has the additional
+ *  side effect of advancing the underlying matcher to the next unconsumed match.
+ *  This effect is visible in the `MatchData` representing the "current match".
  *
  *  {{{
- *  val num = """(\d+)""".r
- *  val all = num.findAllIn("123").toList  // List("123"), not List("123", "23", "3")
+ *  val r = "(ab+c)".r
+ *  val s = "xxxabcyyyabbczzz"
+ *  r.findAllIn(s).start    // 3
```

Comment:
This is a little weird, it works under scala2.12.1, but output the following error under scala2.11.8


>scala> r.findAllIn(s).start
java.lang.IllegalStateException: No match available
  at java.util.regex.Matcher.start(Matcher.java:343)
  at scala.util.matching.Regex$MatchIterator.start(Regex.scala:761)
  ... 32 elided